sdk/src/Services/S3/Custom/Transfer/_async/TransferUtility.async.cs

```diff
@@ -208,11 +210,15 @@ public partial class TransferUtility : ITransferUtility
         ///     A cancellation token that can be used by other objects or threads to receive notice of cancellation.
         /// </param>
         /// <returns>The task object representing the asynchronous operation.</returns>
-        public Task UploadAsync(TransferUtilityUploadRequest request, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task UploadAsync(TransferUtilityUploadRequest request, CancellationToken cancellationToken = default(CancellationToken))
         {
-            CheckForBlockedArn(request.BucketName, "Upload");
-            var command = GetUploadCommand(request, null);
-            return command.ExecuteAsync(cancellationToken);
+            using (var span = AWSConfigs.TelemetryProvider.TracerProvider.GetTracer(S3TransferTracerScope)
+                .CreateSpan($"{nameof(TransferUtility)}.{nameof(UploadAsync)}", null, Amazon.Runtime.Telemetry.Tracing.SpanKind.CLIENT))
```

Comment:
There are some common attributes SDKs should add for the traces, which can be found [in this utility](https://github.com/aws/aws-sdk-net/blob/main/sdk/src/Core/Amazon.Runtime/Telemetry/Tracing/TracingUtilities.cs#L47). Unfortunately I didn't account for it to be called without the request context, but feel free to refactor it or create a different one that fits your needs.

Reply:
Fixed